src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +16,42 @@
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        Session session = null;
+        Transaction transaction = null;
+
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+            return movie;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cant add movie to DB", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
```

Comment:
Similar to the `add` method, retrieving the `SessionFactory` from `HibernateUtil` is redundant. Use the `sessionFactory` field from the superclass to maintain consistency and avoid potential issues.